internal/app/networkd/pkg/nic/nic.go

```diff
@@ -2,75 +2,323 @@
+	"github.com/jsimonetti/rtnetlink/rtnl"
+	"github.com/mdlayher/netlink"
 
 	"github.com/talos-systems/talos/internal/app/networkd/pkg/address"
 	"github.com/talos-systems/talos/internal/pkg/kernel"
 	"github.com/talos-systems/talos/pkg/constants"
+	"github.com/talos-systems/talos/pkg/retry"
 )
 
-// TODO: Probably should put together some map here
-// to map number with string()
 const (
-	Bond = iota
-	Single
-
 	// https://tools.ietf.org/html/rfc791
 	MinimumMTU = 68
 	MaximumMTU = 65536
 )
 
 // NetworkInterface provides an abstract configuration representation for a
-// network interface
+// network interface.
 type NetworkInterface struct {
 	Ignore        bool
 	Name          string
 	Type          int
+	Bonded        bool
 	MTU           uint32
-	Index         uint32
-	SubInterfaces []string
+	Link          *net.Interface
+	SubInterfaces []*net.Interface
 	AddressMethod []address.Addressing
-}
+	BondSettings  []netlink.Attribute
 
-// IsIgnored checks the network interface to see if it should be ignored and not configured
-func (n *NetworkInterface) IsIgnored() bool {
-	if n.Ignore || kernel.ProcCmdline().Get(constants.KernelParamNetworkInterfaceIgnore).Contains(n.Name) {
-		return true
-	}
-
-	return false
+	rtConn   *rtnetlink.Conn
+	rtnlConn *rtnl.Conn
 }
 
-// Create returns a NetworkInterface with all of the given setter options
-// applied
-func Create(link *net.Interface, setters ...Option) (*NetworkInterface, error) {
+// New returns a NetworkInterface with all of the given setter options applied.
+func New(setters ...Option) (*NetworkInterface, error) {
 	// Default interface setup
 	iface := defaultOptions()
 
 	// Configure interface with any specified options
 	var result *multierror.Error
 	for _, setter := range setters {
-		result = multierror.Append(setter(iface))
+		result = multierror.Append(result, setter(iface))
 	}
 
 	// TODO: May need to look at switching this around to filter by Interface.HardwareAddr
 	// Ensure we have an interface name defined
 	if iface.Name == "" {
-		result = multierror.Append(errors.New("interface must have a name"))
+		result = multierror.Append(result, errors.New("interface must have a name"))
 	}
 
 	// If no addressing methods have been configured, default to DHCP
 	// TODO: do we want this behavior or to be explicit with config
 	// so we dont configure every interface be default?
-	if len(iface.AddressMethod) == 0 {
-		iface.AddressMethod = append(iface.AddressMethod, &address.DHCP{NetIf: link})
+
+	// TODO readd this later when we've refactored address to not require a link here?
+	/*
+		if len(iface.AddressMethod) == 0 {
+			iface.AddressMethod = append(iface.AddressMethod, &address.DHCP{NetIf: link})
+		}
+	*/
+
+	// Handle netlink connection
+	conn, err := rtnl.Dial(nil)
+	if err != nil {
+		result = multierror.Append(result, err)
+		return nil, result.ErrorOrNil()
 	}
+	iface.rtnlConn = conn
+
+	// Need rtnetlink for MTU setting
+	// TODO: possible rtnl enhancement
```

Comment:
I'm curious, what is the `rtnl enhancement` here?

Reply:
Right now our rtnetlink/rtnl usage/distinction is a bit clunky. 
`rtnl` provides some high level abstractions when working with the interfaces so you don't need to deal with the underlying netlink message ( ex, `rtnl.LinkUp()` ). Unfortunately the `rtnl` package is pretty limited in it's functionality. There's the potential that we can push a bunch of things upstream so we can make use of `rtnl` exclusively, but I haven't started those conversations yet.